hil/cli/project.py

```diff
@@ -80,6 +82,42 @@ def project_connect_node(project, node):
 @project_node.command(name='remove')
 @click.argument('project')
 @click.argument('node')
-def project_detach_node(project, node):
+@click.option('--force', is_flag=True)
+def project_detach_node(project, node, force):
     """Remove <node> from <project>"""
+    if force:
+        client.node.disable_obm(node)
+        print("Disabled OBM")
+        node_info = client.node.show(node)
+        statuses = []
+        if 'nics' in node_info:
+            for n in node_info['nics']:
+                # if any nic is connected to a network, then it has a switch
+                # and port. So call revert port on those.
+                if 'networks' in n:
+                    try:
+                        response = client.port.port_revert(
+                                                    n['switch'], n['port'])
+                        statuses.append(response['status_id'])
+                    except KeyError:
+                        print("Only admins can force remove nodes.")
+                        sys.exit(1)
+
+        done = False
+        end_time = time.time() + HIL_TIMEOUT
+        while not done:
+            if time.time() > end_time:
+                raise Exception("Removing networks is taking too long. "
```

Comment:
I was worried about the case where, let's say, the networking server is not responding, so the action will always be pending. The user wouldn't know why this call is blocked since it will be looping forever; which is why I thought of setting a time limit.

Reply:
I agree that just having this as a fixed timeout to start is the right thing. If we find it's too short we can make it longer. We could make it configurable, but it's not obvious to me that there are likely use cases for tuning it; let's wait until they actually arise.